src/rollbacks.rs

```diff
@@ -51,16 +64,24 @@ impl<T: Clone> RollbackableVec<T> {
             entries: Vec::new(),
         }
     }
+
+    pub fn get_logs_after_snapshot(
+        &self,
+        snapshot: <RollbackableVec<T> as Rollbackable>::Snapshot,
+    ) -> &[T] {
+        &self.entries[snapshot..]
+    }
 }
 
 impl<T: Clone> Rollbackable for RollbackableVec<T> {
-    type Snapshot = Vec<T>;
+    type Snapshot = usize;
```

Comment:
Let's add some docs/comments on why this is a size, I understand that this is an index to the snapshot, but let's do it for clarity's sake.